FIX-2485 Query data grid

eliasbruvik opened the pull request:
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2485 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

Added a "Data Grid" toggle that changes the view between the query editor and a data grid view.
The data grid view shows the value in an editable textfield, and the documentation for the value (as specified in the xsd documents).

Any row that is selected (as marked by a checkbox and a fully marked row) is included in the query.

All objects we normally use in Witsml Explorer is supported in addition to a few extra. However, not all objects are supported yet. These can be added at a later point if requested.

![image](https://github.com/user-attachments/assets/105696ec-bd3f-465a-af7b-3607441dbea0)


## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [x] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Desktop
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

eliasbruvik committed:
FIX-2485 data grid initial draft

eliasbruvik committed:
merge template and query to form the table data

eliasbruvik committed:
fix initially selected rows

eliasbruvik committed:
fix row selection change + query generation from table data

eliasbruvik committed:
fix initially selected rows

eliasbruvik committed:
fix tab change

eliasbruvik committed:
add input field

eliasbruvik committed:
update input

eliasbruvik committed:
use controlled state

eliasbruvik committed:
fix row count for nested data

eliasbruvik committed:
fix proper inset for nested objects

eliasbruvik committed:
Fix reset order

eliasbruvik committed:
Fix table height

eliasbruvik committed:
update template format & add bhaRun

eliasbruvik committed:
add changelog

eliasbruvik committed:
add fluidsReport

eliasbruvik committed:
use common properties

eliasbruvik committed:
add formationMarker

eliasbruvik committed:
add log

eliasbruvik committed:
Add cementJob

eliasbruvik committed:
Add wbGeometry

eliasbruvik committed:
Move common wbGeometry properties

eliasbruvik committed:
Add convCore

eliasbruvik committed:
Add Risk

eliasbruvik committed:
Add MudLog

eliasbruvik committed:
Add Message

eliasbruvik committed:
Add Rig

eliasbruvik committed:
Add Trajectory

eliasbruvik committed:
add todo

eliasbruvik committed:
Add Tubular

eliasbruvik committed:
Add Well

eliasbruvik committed:
Add Wellbore

eliasbruvik committed:
cleanup

robertbasti commented:
When clicking on Pick template and choosing drillReport - it shows "Unable to parse query".  Could not be a better message?

robertbasti commented on Src/WitsmlExplorer.Frontend/components/ContentViews/QueryDataGrid.tsx:
```diff
@@ -0,0 +1,363 @@
+      />
+    </div>
+  ) : (
+    <Typography>Unable to parse query</Typography>
```
cound not be there a more descriptive message?

robertbasti replied:
e.g. This object is not supported yet.

robertbasti commented:
when removing e.g. xmlns or version and putting back for example for attachment or other templates, it returns 
Error while querying store: -468 - A queryIn template must include a version attribute in the plural data-object that defines the Data Schema Version of the data-object. It removes the value. Is that desirable behavior?

robertbasti reviewed (dismissed):
code seems ok

eliasbruvik committed:
Update parse warning

eliasbruvik commented:
> When clicking on Pick template and choosing drillReport - it shows "Unable to parse query". Could not be a better message?

Updated the warning.

> when removing e.g. xmlns or version and putting back for example for attachment or other templates, it returns
Error while querying store: -468 - A queryIn template must include a version attribute in the plural data-object that defines the Data Schema Version of the data-object. It removes the value. Is that desirable behavior?

I guess this is intentional (at least for now). If you remove a value from the dataGrid it's also removed in the query. So to get it back you would have to re-specify the values.

robertbasti reviewed (approved):
tested and seems ok
